src/back/msl.rs

```diff
@@ -269,41 +285,49 @@ impl<'a> TypedGlobalVariable<'a> {
     }
 }
 
-impl Display for ResolvedBinding {
-    fn fmt(&self, formatter: &mut Formatter<'_>) -> Result<(), FmtError> {
+impl ResolvedBinding {
+    fn try_fmt<W: Write>(&self, formatter: &mut W) -> Result<(), Error> {
         match *self {
             ResolvedBinding::BuiltIn(built_in) => {
                 let name = match built_in {
                     spirv::BuiltIn::ClipDistance => "clip_distance",
                     spirv::BuiltIn::GlobalInvocationId => "thread_position_in_grid",
                     spirv::BuiltIn::PointSize => "point_size",
                     spirv::BuiltIn::Position => "position",
-                    _ => panic!("Built in {:?} is not implemented", built_in),
+                    _ => return Err(Error::UnimplementedBuiltIn(built_in)),
                 };
-                formatter.write_str(name)
+                Ok(formatter.write_str(name)?)
```

Comment:
we shouldn't need that syntax: `Ok(something?)`, unless it's deliberate

Reply:
Ah, okay. How would you prefer this line be written?

Reply:
Actually, I was wrong here. Sorry!
The `?` operator can convert between error types, but just calling `write_str` would not.

Reply:
Yeah, that tripped me up at first as well! 😂